blob/blob.go

```diff
@@ -61,51 +59,56 @@ func (p Proof) equal(input Proof) error {
 
 // Blob represents any application-specific binary data that anyone can submit to Celestia.
 type Blob struct {
-	*blob.Blob `json:"blob"`
+	*gosquare.Blob `json:"blob"`
 
 	Commitment Commitment `json:"commitment"`
 
-	// the celestia-node's namespace type
-	// this is to avoid converting to and from app's type
-	namespace share.Namespace
-
 	// index represents the index of the blob's first share in the EDS.
 	// Only retrieved, on-chain blobs will have the index set. Default is -1.
 	index int
 }
 
 // NewBlobV0 constructs a new blob from the provided Namespace and data.
 // The blob will be formatted as v0 shares.
-func NewBlobV0(namespace share.Namespace, data []byte) (*Blob, error) {
-	return NewBlob(appconsts.ShareVersionZero, namespace, data)
+func NewBlobV0(namespace gosquare.Namespace, data []byte) (*Blob, error) {
+	return NewBlob(gosquare.ShareVersionZero, namespace, data, nil)
 }
 
-// NewBlob constructs a new blob from the provided Namespace, data and share version.
-func NewBlob(shareVersion uint8, namespace share.Namespace, data []byte) (*Blob, error) {
+// NewBlobV1 constructs a new blob from the provided Namespace, data, and signer.
+// The blob will be formatted as v1 shares.
+func NewBlobV1(namespace gosquare.Namespace, data, signer []byte) (*Blob, error) {
```

Comment:
a v1 blob contains also a signer. So, having NewBlob constructs only the latest blob types is not the case because new version can have new additional fields

Reply:
but we can have NewBlob + applied options

Reply:
The main reason to introduce a versioned constructor is devex. You don't want to import a constant from the app every time you need to make a share, a blob, or a namespace. Also, each version may have different set of params, like V0 Blob doesn't have a signer and V1 does, so you cant' necessary generalize the constructor. 

Furthermore, I would argue that users should be aware of its version. Depending always on the last version may lead to accidental breakages for the blob-consuming code after the type version bump.  I think its useful might be to provide `NewBlob` option that always points to latest version, but only along other versioned constructors.